R/getShareECprojections.R

```diff
@@ -230,17 +252,26 @@ getShareECprojections <- function(config,
                   .data[["shareEndHist"]]),
              .data[["obj_share"]]
            )) %>%
+    left_join(hdd, by = "region") %>%
+    mutate(obj_share = ifelse(
+             .data[["variable"]] %in% useCarrierDH & .data[["hdd"]] < thresDh,
+             .data[["shareEndHist"]],
+             .data[["obj_share"]]
```

Comment:
We always want to avoid decreasing DH shares, right? I actually though this was already the case but couldn't find it. This change should do this. I hope...
```suggestion
             .data[["variable"]] %in% useCarrierDH,
             pmax(.data[["obj_share"]], ifelse(.data[["hdd"]] < thresDh, 0, .data[["shareEndHist"]])),
             .data[["obj_share"]]
```

Reply:
Okay, above, I could only find the upper limit of 0.5 the global target share, not the non-decreasing condition. But maybe I missed it.

Reply:
This bit of code (lines 249-254) should ensure this:
```
mutate(enduse = gsub(".[a-z]*$", "", .data[["variable"]]),
           obj_share = ifelse(
             .data[["variable"]] %in% useCarrierInert,
             pmax(.data[["obj_share"]]* ifelse(.data[["shareEndHist"]] <= thresInert, 0.5, 1),
                  .data[["shareEndHist"]]),
             .data[["obj_share"]]
           ))
```

Reply:
Oh yeah, stupid me. Thanks!